set_fm_data.py

```diff
@@ -75,6 +78,11 @@ class SetFMdata(hass.Hass, WallboxModbusMixin):
 
     
     def initialize(self):
+        # ToDo: AJO 2022-12-30: This code is copied in several modules: combine!
+        self.CAR_MIN_SOC_IN_PERCENT = int(float(self.args["car_min_soc_in_percent"]))
+        # Make sure this value is between 10 en 30
+        self.CAR_MIN_SOC_IN_PERCENT = max(min(30, self.CAR_MIN_SOC_IN_PERCENT), 10)
```

Comment:
This silently overrides a user setting, which I don't particularly like. At the very least it warrants a warning upon initializing the app, and a comment in the readme where the config setting is explained. The used minimum might also be visualized as a horizontal line in the dashboard chart, but this is probably more intricate work.

I could also imagine that there are users that want to configure a minimum SoC higher than 30%.

Reply:
True that it is silently doing this, maybe a notification would be appropriate when it is overwritten.
I also agree that 30 is arbitrary and maybe a bit low, would 50% be oke?

The readme on the secrets.yaml (where this setting is stored) is:
# What would you like to be the minimum charge in your battery?
# The scheduling will not discharge below this value and if the car returns with
# and SoC below this value, the battery will be charged to this minimum asap
# before regular scheduling.
# A high value results in always having a greater driving range available, even 
# when not planned, but less capacity available for dis-charge and so lesser
# earnings.
# A lower value results in sometimes a smaller driving range available for
# un-planned drives but there is always more capacity for discharge and so more
# earnings.
# Some research suggests battery life is shorter if min SoC is below 15%.
# In some cars the SoC every now and then skips a number, eg. from 21 to 19%, 
# skipping 20%. This might result in toggling charging behaviour around this
# minimum SoC. If this happens try a value 1 higher or lower.
# The setting must be an integer between 10 and 30%, default is 20%.
car_min_soc_in_percent: 20

Reply:
I personally wouldn't enforce an upper limit for this user setting, but as long as it's clearly documented I'm okay.

Reply:
I added a notification for the user (persistent: shows in UI as orange dot on menu-item "Notifications"), can be dismissed.
I think an upper limit is relevant, a setting above say 70% would make the system useless i guess..
Also: we plan to add a max soc in percent. This setting must always be 20% (?) lower than that max, otherwise the system becomes useless. 
So i am keeping it at 30% for now and will change to max_soc_in_percent - 20 later.